### PR TITLE
Changed default labels for sdg, tdg, sx, and sxdg to None

### LINKED ISSUE #4883: Name change for S_dg and T_dg not reflected in drawers
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.8
- **Operating system**: Ubuntu 18.04

### What is the current behavior?

Previously the s-dagger and t-dagger gates were named sdg and tdg. This was changed to S_dg and T_dg, which means the drawers no longer display the dagger icon.

### Steps to reproduce the problem

```
qc = QuantumCircuit(1)
qc.sdg(0)
qc.draw('mpl')
```
### What is the expected behavior?

Both gates should display the dagger icon after S or T.

### Suggested solutions

This can be fixed in qcstyle.py.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fix #4883

### Details and comments

Changed default labels for sdg, tdg, sx, and sxdg to None.

### CODE REVIEW

**Comment by ajavadia** [comment]
This is not an mpl issue, I think we should just revert the changes to the gate labels. I don't understand why they happened in the first place.
Ultimately these gates should show up as `S†`, `T†` and `√X†` in text, latex and mpl.

**Comment by enavarro51** [comment]
Ok. I'll remove the default labels for sdg, tdg , sx and sxdg and leave qcstyle as is.

**Comment by enavarro51** [comment]
Removing the labels fixes the problem for mpl and latex. 

For text, I believe @Cryoris had a problem because the dagger was not in code page 437, hence the s_dg. When labels are removed the 4 gates appear in text drawer as SDG, TDG, SX, and SXDG. Previously sdg and tdg showed as Sdg and Tdg in the text drawer. Right now this fails a text drawer test, since S_dg and T_dg were put in the test file. The test file shows sx as √X and sxdg as √X_dg.

So what to do?

**Comment by enavarro51** [comment]
@ajavadia I think this should be solved in the text drawer and the text drawer test. Gate names were changed at some point to display all caps in the text drawer - RX, ISWAP, etc., so SDG and TDG are correct. It's possible to treat the √X and the √X_dg, or maybe √XDG, as special cases in the text drawer.

**Comment by enavarro51** [comment]
```
     ┌─────┐┌─────┐┌────┐┌──────┐
q_0: ┤ SDG ├┤ TDG ├┤ √X ├┤ √XDG ├
     └─────┘└─────┘└────┘└──────┘
q_1: ────────────────────────────
                                 
```
image

File: test/python/visualization/references/circuit_text_ref_original2.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+        ┌───┐┌───┐┌───┐ ░ ┌───┐┌───┐┌─────┐┌───┐┌─────┐┌────┐┌──────┐┌───┐     ┌────────┐┌────────┐┌────────┐┌────────┐┌───────────┐┌──────────────┐                                                                                ┌─┐
```
[the remark this answers is not in the document]
Comment (mtreinish) [reply test/python/visualization/references/circuit_text_ref_original2.txt:1]: Actually I don't think we should add an extra file like this that is not used anywhere. I'm going to remove it and then approve this PR (we're staging a quick 0.15.1 to get #4888 out and want to include this too).

**Pull request merged**